excercises/2-locators/locators-element-all.spec.ts

```diff
@@ -1,37 +1,40 @@
 import { browser, element, by } from 'protractor'
+import { getLeadingCommentRanges } from 'typescript';
 
 describe('Protractor workshop app', function () {
 
 	beforeAll(async function () {
 		await browser.get("/jak-to-zrobic-w-js");
 	});
-
-	/**
-	 * Check http://angular.github.io/protractor/#/api?view=ElementArrayFinder
-	 * to see how get function can be used
-	 */
 	it('should have menu items with links to "Home", "About", "Services", "Blog", "Contact",  pages', async function () {
 		const menuText = ["Home", "About", "Services", "Blog", "Contact"]
-		await browser.get("/jak-to-zrobic-w-js");
-		var menuItems = element.all(by.css('ul.nav > li > a'));
-		expect(await menuItems.get(0).getText()).toEqual(menuText[0]);
+		let menuItems = element.all(by.css('ul.nav > li > a')).map((el)=>{
+			return el.getText();
+		})
+		expect(await menuItems).toEqual(menuText);
 	});
 
-	/**
-	* Check http://angular.github.io/protractor/#/api?view=ElementArrayFinder
-	* to see how map function can be used to verify content of multiple elements
-	*/
-
-	xit('should have Feature A, Feature B, Feature C sections ...', function () {
-		//TODO: User map function to verify feature names
+	it('should have Feature A, Feature B, Feature C sections ...', async function () {
+		const FeatureArray = ["Feature A", "Feature B", "Feature C"];
+		let featureElements = element.all(by.xpath("//h2")).map((el) => {
+		  return el.getText();
+		});
+		expect( await featureElements).toEqual(FeatureArray);
 	});
 
-	/**
-	 * Check http://angular.github.io/protractor/#/api?view=ElementArrayFinder
-	 * to see how fileter function can be used select elements based on condition
-	 */
-	xit('should route to "Blog" pages after selecting link', function () {
-		//TODO: User filter function to verify feature names
+	it('should route to "Blog" pages after selecting link', async function () {
```

Comment:
To be fair I don't even know what is the goal of this test.
I would even accept onlynames function on its own and then using it in filter but:
1. The function is checking something so 'onlynames' may not be best name. Function names should be camel case so 'onlyNames', I would name it 'compareName' or something.
2. You are comparing Author Name with Author Name so the result of filter function makes no difference to the array.
3. The mechanism on its own is ok for me, and as I don't know what that test should do